language/dotnet/run-tests.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "Run your tests"
+keywords: .NET, build, test
+description: How to build and run your tests
+---
+
+{% include_relative nav.html selected="4" %}
+
+## Prerequisites
+
+Work through the steps to build an image and run it as a containerized application in [Use containers for development](develop.md).
+
+## Introduction
+
+Testing is an essential part of modern software development. In combination with 3rd party frameworks or services, Docker helps to test applications without mocks or complicated environment configurations fast and reliably.
+
+## Add .NET test project
+
+To test our sample application, we create a standalone test project from a template using the .NET CLI. On your local machine, open a terminal, change the directory to the `dotnet-docker` directory and run the following command:
+
+```console
+$ cd /path/to/dotnet-docker
+$ dotnet new xunit -n myWebApp.Tests -o tests
+```
+
+Next, we'll update the test project and add the Testcontainers for .NET package that allows us to run tests against Docker resources. Switch to the `tests` directory and run the following command:
+
+```console
+$ dotnet add package Testcontainers
+```
+
+## Add a test
+
+Open the test project in your favorite IDE and replace the contents of `UnitTest1` with the following code:
+
+```c#
+using System.Net;
+using DotNet.Testcontainers.Builders;
+using DotNet.Testcontainers.Containers;
+using DotNet.Testcontainers.Networks;
+
+public sealed class UnitTest1 : IAsyncLifetime, IDisposable
+{
+    private const ushort HttpPort = 80;
+
+    private readonly CancellationTokenSource _cts = new(TimeSpan.FromMinutes(1));
+
+    private readonly IDockerNetwork _network;
+
+    private readonly IDockerContainer _dbContainer;
+
+    private readonly IDockerContainer _appContainer;
+
+    public UnitTest1()
+    {
+        _network = new TestcontainersNetworkBuilder()
+            .WithName(Guid.NewGuid().ToString("D"))
+            .Build();
+
+        _dbContainer = new TestcontainersBuilder<TestcontainersContainer>()
+            .WithImage("postgres")
+            .WithNetwork(_network)
+            .WithNetworkAliases("db")
+            .WithVolumeMount("postgres-data", "/var/lib/postgresql/data")
+            .Build();
+
+        _appContainer = new TestcontainersBuilder<TestcontainersContainer>()
```

Comment:
`TestcontainersBuilder<>`  is disposable and thus should be disposed, how should one handle that?

Reply:
No, it is not necessary. The Resource Reaper will take care of it.